presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreConfig.java

```diff
@@ -169,4 +176,69 @@ public ThriftMetastoreConfig setMaxWaitForTransactionLock(Duration maxWaitForTra
         this.maxWaitForTransactionLock = maxWaitForTransactionLock;
         return this;
     }
+
+    public boolean isTlsEnabled()
+    {
+        return tlsEnabled;
+    }
+
+    @Config("hive.metastore.thrift.client.tls.enabled")
+    @ConfigDescription("Whether TLS security is enabled")
+    public ThriftMetastoreConfig setTlsEnabled(boolean tlsEnabled)
+    {
+        this.tlsEnabled = tlsEnabled;
+        return this;
+    }
+
+    public File getKeystorePath()
+    {
+        return keystorePath;
+    }
+
+    @Config("hive.metastore.thrift.client.tls.keystore.path")
+    @ConfigDescription("Path to the PEM or JKS key store")
+    public ThriftMetastoreConfig setKeystorePath(File keystorePath)
+    {
+        this.keystorePath = keystorePath;
```

Comment:
Is validation upstream or downstream that this `File` exists?  (Not sure what the contract for configuration loading is and where validation is expected.)

Reply:
The validation is done by `try (InputStream in = new FileInputStream(keyStorePath.get())) {}` and `try (InputStream in = new FileInputStream(trustStorePath)) {}` in downstream